lib/services/internalEngine/index.js

```diff
@@ -571,6 +570,41 @@ class InternalEngine {
     })
       .catch(error => Bluebird.reject(this.esWrapper.formatESError(error)));
   }
+
+  /**
+   * Apply default mapping to the collection but preserve existing kuzzle metadata
+   *
+   * @param {string} index
+   * @param {string} collection
+   * @returns {Promise.<object>} returns the updated default mapping
+   */
+  applyDefaultMapping (index, collection, mapping) {
+    const defaultMapping = mapping;
+
+    return this.getMapping({ index, type: collection }, true)
+      .then(mappings => {
+        const existingMapping = mappings[index].mappings[collection];
+
+        // Preserve old version of kuzzle metadata mapping
+        if (existingMapping.properties && existingMapping.properties._kuzzle_info) {
+          Object.assign(
+            defaultMapping._kuzzle_info.properties,
+            existingMapping.properties._kuzzle_info.properties
+          );
+        }
+
+        const updatedMapping = {
+          [collection]: {
+            dynamic: existingMapping.dynamic || this.config.dynamic,
```

Comment:
In this case, I think that the configuration should take precedence over the existing dynamic mapping setting: at any point of time, the configuration file represents the "truth" about how Kuzzle should behave, while previously set configurations might now be obsolete.
So you should reverse that test.

Reply:
If i change the precedence here, Kuzzle will apply his default dynamic policy when he restart.  
Users might want to have a default dynamic mapping policy for every collection and a different policy for just one collection

Reply:
I agree with Adrien's point of view